src/dstack/_internal/core/backends/runpod/compute.py

```diff
@@ -0,0 +1,146 @@
+    InstanceOfferWithAvailability,
+    LaunchedInstanceInfo,
+    SSHKey,
+)
+from dstack._internal.core.models.runs import Job, Requirements, Run
+from dstack._internal.utils.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+class RunpodCompute(Compute):
+    def __init__(self, config):
+        self.config = config
+        self.api_client = RunpodApiClient(config.creds.api_key)
+
+    def get_offers(
+        self, requirements: Optional[Requirements] = None
+    ) -> List[InstanceOfferWithAvailability]:
+        offers = get_catalog_offers(
+            backend=BackendType.RUNPOD,
+            requirements=requirements,
+        )
+        offers = [
+            InstanceOfferWithAvailability(
+                **offer.dict(), availability=InstanceAvailability.AVAILABLE
+            )
+            for offer in offers
+        ]
+        return offers
+
+    def run_job(
+        self,
+        run: Run,
+        job: Job,
+        instance_offer: InstanceOfferWithAvailability,
+        project_ssh_public_key: str,
+        project_ssh_private_key: str,
+    ) -> LaunchedInstanceInfo:
+        instance_config = InstanceConfiguration(
+            project_name=run.project_name,
+            instance_name=get_instance_name(run, job),
+            ssh_keys=[
+                SSHKey(public=run.run_spec.ssh_key_pub.strip()),
+                SSHKey(public=project_ssh_public_key.strip()),
+            ],
+            job_docker_config=None,
+            user=run.user,
+        )
+        launched_instance_info = self.create_instance(instance_offer, instance_config)
+        return launched_instance_info
+
+    def create_instance(
+        self,
+        instance_offer: InstanceOfferWithAvailability,
+        instance_config: InstanceConfiguration,
+    ) -> LaunchedInstanceInfo:
+        authorized_keys = instance_config.get_public_keys()
+        memory_size = round(instance_offer.instance.resources.memory_mib / 1024)
+        disk_size = round(instance_offer.instance.resources.disk.size_mib / 1024)
+        resp = self.api_client.create_pod(
+            name=instance_config.instance_name,
+            image_name="runpod/pytorch:2.1.1-py3.10-cuda12.1.1-devel-ubuntu22.04",
+            gpu_type_id=instance_offer.instance.name,
+            cloud_type="ALL",  # ["ALL", "COMMUNITY", "SECURE"]:
+            gpu_count=len(instance_offer.instance.resources.gpus),
+            container_disk_in_gb=disk_size,
+            min_vcpu_count=instance_offer.instance.resources.cpus,
+            min_memory_in_gb=memory_size,
+            support_public_ip=True,
+            docker_args=get_docker_args(authorized_keys),
+            ports="22/tcp",
+        )
+
+        instance_id = resp["id"]
+        # Wait until VM State is Active. This is necessary to get the ip_address.
+        pod = self.api_client.get_pod(instance_id)
+        while pod["runtime"] is None:
+            pod = self.api_client.get_pod(resp["id"])
+            time.sleep(1)
```

Comment:
Potentially an infinite loop. There should be a limit on the number of attempts.

Reply:
Ok I will check the average response time runpod takes,  then add a margin to set number of attempts.